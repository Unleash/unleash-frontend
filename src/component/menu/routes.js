import CreateFeatureToggle from '../../page/features/create';
import CopyFeatureToggle from '../../page/features/copy';
import ViewFeatureToggle from '../../page/features/show';
import Features from '../../page/features';
import CreateStrategies from '../../page/strategies/create';
import StrategyView from '../../page/strategies/show';
import Strategies from '../../page/strategies';
import HistoryPage from '../../page/history';
import HistoryTogglePage from '../../page/history/toggle';
import ShowArchive from '../../page/archive/show';
import Archive from '../../page/archive';
import Applications from '../../page/applications';
import ApplicationView from '../../page/applications/view';
import ContextFields from '../../page/context';
import CreateContextField from '../../page/context/create';
import EditContextField from '../../page/context/edit';
import LogoutFeatures from '../../page/user/logout';

export const routes = [
    // Features
    { path: '/features/create', parent: '/features', title: 'Create', component: CreateFeatureToggle },
    { path: '/features/copy/:copyToggle', parent: '/features', title: 'Copy', component: CopyFeatureToggle },
    { path: '/features/:activeTab/:name', parent: '/features', title: ':name', component: ViewFeatureToggle },
    { path: '/features', title: 'Feature Toggles', icon: 'list', component: Features },

    // Strategies
    { path: '/strategies/create', title: 'Create', parent: '/strategies', component: CreateStrategies },
    {
        path: '/strategies/:activeTab/:strategyName',
        title: ':strategyName',
        parent: '/strategies',
        component: StrategyView,
    },
    { path: '/strategies', title: 'Strategies', icon: 'extension', component: Strategies },

    // History
    { path: '/history/:toggleName', title: ':toggleName', parent: '/history', component: HistoryTogglePage },
    { path: '/history', title: 'Event History', icon: 'history', component: HistoryPage },

    // Archive
    { path: '/archive/:activeTab/:name', title: ':name', parent: '/archive', component: ShowArchive },
    { path: '/archive', title: 'Archived Toggles', icon: 'archive', component: Archive },

    // Applications
    { path: '/applications/:name', title: ':name', parent: '/applications', component: ApplicationView },
    { path: '/applications', title: 'Applications', icon: 'apps', component: Applications },

    // Context
    { path: '/context/create', parent: '/context', title: 'Create', component: CreateContextField },
    { path: '/context/edit/:name', parent: '/context', title: ':name', component: EditContextField },
    { path: '/context', title: 'Context Fields', icon: 'apps', component: ContextFields, hidden: true },
];

export const getRoute = path => routes.find(route => route.path === path);

export const baseRoutes = routes.filter(route => !route.hidden).filter(route => !route.parent);
