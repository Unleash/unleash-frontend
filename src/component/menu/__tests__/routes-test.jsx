import { routes, baseRoutes, getRoute } from '../routes';

test('returns all defined routes', () => {
    expect(routes.length).toEqual(14);
    expect(routes).toMatchSnapshot();
});

test('returns all baseRoutes', () => {
    expect(baseRoutes.length).toEqual(6);
    expect(baseRoutes).toMatchSnapshot();
});

test('getRoute() returns named route', () => {
    const featuresRoute = getRoute('/features');
    expect(featuresRoute.path).toEqual('/features');
});
