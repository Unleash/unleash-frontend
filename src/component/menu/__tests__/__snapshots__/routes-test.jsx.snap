// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all baseRoutes 1`] = `
Array [
  Object {
    "component": [Function],
    "icon": "list",
    "path": "/features",
    "title": "Feature Toggles",
  },
  Object {
    "component": [Function],
    "icon": "extension",
    "path": "/strategies",
    "title": "Strategies",
  },
  Object {
    "component": [Function],
    "icon": "history",
    "path": "/history",
    "title": "Event History",
  },
  Object {
    "component": [Function],
    "icon": "archive",
    "path": "/archive",
    "title": "Archived Toggles",
  },
  Object {
    "component": [Function],
    "icon": "apps",
    "path": "/applications",
    "title": "Applications",
  },
  Object {
    "component": [Function],
    "flag": "C",
    "icon": "album",
    "path": "/context",
    "title": "Context Fields",
  },
  Object {
    "component": [Function],
    "flag": "P",
    "icon": "folder_open",
    "path": "/projects",
    "title": "Projects",
  },
  Object {
    "component": [Function],
    "icon": "label",
    "path": "/tag-types",
    "title": "Tag types",
  },
  Object {
    "component": [Function],
    "hidden": false,
    "icon": "device_hub",
    "path": "/addons",
    "title": "Addons",
  },
  Object {
    "component": [Function],
    "icon": "report",
    "path": "/reporting",
    "title": "Reporting",
  },
  Object {
    "component": [Function],
    "icon": "exit_to_app",
    "path": "/logout",
    "title": "Sign out",
  },
]
`;

exports[`returns all defined routes 1`] = `
Array [
  Object {
    "component": [Function],
    "parent": "/features",
    "path": "/features/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/features",
    "path": "/features/copy/:copyToggle",
    "title": "Copy",
  },
  Object {
    "component": [Function],
    "parent": "/features",
    "path": "/features/:activeTab/:name",
    "title": ":name",
  },
  Object {
    "component": [Function],
    "icon": "list",
    "path": "/features",
    "title": "Feature Toggles",
  },
  Object {
    "component": [Function],
    "parent": "/strategies",
    "path": "/strategies/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/strategies",
    "path": "/strategies/:activeTab/:strategyName",
    "title": ":strategyName",
  },
  Object {
    "component": [Function],
    "icon": "extension",
    "path": "/strategies",
    "title": "Strategies",
  },
  Object {
    "component": [Function],
    "parent": "/history",
    "path": "/history/:toggleName",
    "title": ":toggleName",
  },
  Object {
    "component": [Function],
    "icon": "history",
    "path": "/history",
    "title": "Event History",
  },
  Object {
    "component": [Function],
    "parent": "/archive",
    "path": "/archive/:activeTab/:name",
    "title": ":name",
  },
  Object {
    "component": [Function],
    "icon": "archive",
    "path": "/archive",
    "title": "Archived Toggles",
  },
  Object {
    "component": [Function],
    "parent": "/applications",
    "path": "/applications/:name",
    "title": ":name",
  },
  Object {
    "component": [Function],
    "icon": "apps",
    "path": "/applications",
    "title": "Applications",
  },
  Object {
    "component": [Function],
    "parent": "/context",
    "path": "/context/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/context",
    "path": "/context/edit/:name",
    "title": ":name",
  },
  Object {
    "component": [Function],
    "flag": "C",
    "icon": "album",
    "path": "/context",
    "title": "Context Fields",
  },
  Object {
    "component": [Function],
    "parent": "/projects",
    "path": "/projects/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/projects",
    "path": "/projects/edit/:id",
    "title": ":id",
  },
  Object {
    "component": [Function],
    "flag": "P",
    "icon": "folder_open",
    "path": "/projects",
    "title": "Projects",
  },
  Object {
    "component": [Function],
    "parent": "/admin",
    "path": "/admin/api",
    "title": "API access",
  },
  Object {
    "component": [Function],
    "parent": "/admin",
    "path": "/admin/users",
    "title": "Users",
  },
  Object {
    "component": [Function],
    "parent": "/admin",
    "path": "/admin/auth",
    "title": "Authentication",
  },
  Object {
    "component": [Function],
    "hidden": true,
    "icon": "album",
    "path": "/admin",
    "title": "Admin",
  },
  Object {
    "component": [Function],
    "parent": "/tag-types",
    "path": "/tag-types/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/tag-types",
    "path": "/tag-types/edit/:name",
    "title": ":name",
  },
  Object {
    "component": [Function],
    "icon": "label",
    "path": "/tag-types",
    "title": "Tag types",
  },
  Object {
    "component": [Function],
    "parent": "/tags",
    "path": "/tags/create",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "hidden": true,
    "icon": "label",
    "path": "/tags",
    "title": "Tags",
  },
  Object {
    "component": [Function],
    "parent": "/addons",
    "path": "/addons/create/:provider",
    "title": "Create",
  },
  Object {
    "component": [Function],
    "parent": "/addons",
    "path": "/addons/edit/:id",
    "title": "Edit",
  },
  Object {
    "component": [Function],
    "hidden": false,
    "icon": "device_hub",
    "path": "/addons",
    "title": "Addons",
  },
  Object {
    "component": [Function],
    "icon": "report",
    "path": "/reporting",
    "title": "Reporting",
  },
  Object {
    "component": [Function],
    "icon": "exit_to_app",
    "path": "/logout",
    "title": "Sign out",
  },
]
`;
